R/Figures.R

```diff
@@ -1127,6 +1127,340 @@ TADA_TwoCharacteristicScatterplot <- function(.data, id_cols = "TADA.ComparableD
         ), "<br>"
       )
     )
+  
+  return(scatterplot)
+}
+  
+#' Create A Scatterplot of Multiple Field Groups within a Column
+#'
+#' @param .data TADA data frame containing the data downloaded from the WQP,
+#'   where each row represents a unique data record. Data frame must include the
+#'   columns 'TADA.ComparableDataIdentifier', 'TADA.ResultMeasureValue', and 'TADA.ResultMeasure.MeasureUnitCode'
+#'   to run this function.
+#'
+#' @param id_cols The column in the dataset used to identify the unique groups to
+#'   be plotted. Defaults to 'TADA.ComparableDataIdentifier'. This input is flexible,
+#'   however, for the specific use case in the TADAShiny app where a user might
+#'   create groups based on a concatenation of the comparable data identifier
+#'   with other additional grouping variables (e.g. site type, site name, year,
+#'   organization, etc.)
+#'
+#' @param groups A vector of two identifiers from the id_cols column. For
+#'   example, if the id_cols is 'TADA.ComparableDataIdentifier', the groups could
+#'   be 'DISSOLVED OXYGEN (DO)_NA_NA_UG/L' and 'PH_NA_NA_NA'. These groups will
+#'   be specific to your dataset. If the id_cols is 'MonitoringLocationName',
+#'   the groups could be 'Upper Red Lake: West' and 'Upper Red Lake: West-Central'.
+#'
+#' @return A single plotly scatterplot figure with one x-axis (Date/Time) and a
+#'   left axis showing the units of a single characteristic groups
+#'   plotted on the same figure area wit a legend for each groups
+#'
+#' @export
+#'
+#' @examples
+#' # Load example dataset:
+#' data(Data_Nutrients_UT)
+#' # Create a single scatterplot with two monitoring locations (groups) for a single TADA.ComparableDataIdentifier. 
+#' # Dataframe has multiple comparable data identifier that was not filtered, select value for 'NITROGEN_TOTAL_AS N_MG/L' to plot this characteristic value.
+#' TADA_MultiScatterplot(Data_Nutrients_UT, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"), groups = c("UTAH LAKE 1 MI EAST OF PELICAN POINT", "Utah Lake 2 miles west of Vineyard"))
+#'
+#' # Load example dataset:
+#' data(Data_6Tribes_5y_Harmonized)
+#' # Filter the example data so it includes only one TADA.ComparableDataIdentifier
+#' df <- dplyr::filter(Data_6Tribes_5y_Harmonized, TADA.ComparableDataIdentifier %in% c("TOTAL PHOSPHORUS, MIXED FORMS_UNFILTERED_AS P_MG/L"))
+#' # Creates a scatterplot of the three specified sites of interest in the same plot.
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"), groups = c("Upper Red Lake: West", "Upper Red Lake: West-Central","Upper Red Lake: East Central"))
+#' # If no groups are selected, return the top 4 groups by MonitoringLocationName field counts.
+#' TADA_MultiScatterplot(df, id_cols = c("TADA.ComparableDataIdentifier", "MonitoringLocationName"))
+#' # Both commands below will return a single scatterplot for just the single comparable data identifier. 
```

Comment:
Should this be included as an example as it is essentially recreating TADA_Scatterplot? Should users running this function be prompted to use TADA_Scatterplot instead if they don't need groups? Or should we retire TADA_Scatterplot (perhaps changing the name of the new TADA_MultiScatterplot function to TADA_Scatterplot?) as this function can handle that use case as well?

This would be a good thing to discuss at a team meeting before we make a decision either way.

Reply:
I was thinking the same. It would just provide mainly the same output as TADA_Scatterplot at that point, except with a single legend option. It would be like an alternative function to use besides TADA_Scatterplot

Reply:
Maybe you can demo the no-group output of your new function and the output of the original TADA_Scatterplot in our team meeting? That way we can all be on the same page to make a decision.

Reply:
Yes, I can demo this at the team meeting for us to decide